packages/core2/src/managers/_RepositoryManager.ts

```diff
@@ -1,6 +1,255 @@
+import * as t from "io-ts";
+import fetch, { Response } from "node-fetch";
+
 import { BaseManager } from "./_BaseManager";
 
+import { APIEndpoints, SLICE_MACHINE_USER_AGENT } from "../constants";
+import { decode } from "../lib/decode";
+import { serializeCookies } from "../lib/serializeCookies";
+
+const DEFAULT_REPOSITORY_SETTINGS = {
+	plan: "personal",
+	isAnnual: "false",
+	role: "developer",
+};
+
+const RepositoryUserAgents = {
+	SliceMachine: "prismic-cli/sm",
+	LegacyZero: "prismic-cli/0",
+} as const;
+type RepositoryUserAgents =
+	typeof RepositoryUserAgents[keyof typeof RepositoryUserAgents];
+
+const PrismicRepositoryRoles = {
+	SuperUser: "SuperUser",
+	Administrator: "Administrator",
+	Owner: "Owner",
+	Manager: "Manager",
+	Publisher: "Publisher",
+	Writer: "Writer",
+	Readonly: "Readonly",
+} as const;
+type PrismicRepositoryRoles =
+	typeof PrismicRepositoryRoles[keyof typeof PrismicRepositoryRoles];
+
+const PrismicRepository = t.type({
+	domain: t.string,
+	name: t.string,
+	role: t.union([
+		t.keyof(PrismicRepositoryRoles),
+		t.record(t.string, t.keyof(PrismicRepositoryRoles)),
+	]),
+});
+export type PrismicRepository = t.TypeOf<typeof PrismicRepository>;
+
+type SliceMachineManagerExistsRepositoryArgs = {
+	domain: string;
+};
+
+type SliceMachineManagerCreateRepositoryArgs = {
+	domain: string;
+	framework: string; // TODO: Type(?)
+};
+
+type SliceMachineManagerDeleteRepositoryArgs = {
+	domain: string;
+	password: string;
+};
+
+type SliceMachineManagerPushDocumentsArgs = {
+	domain: string;
+	signature: string;
+	documents: Record<string, unknown>; // TODO: Type unknown if possible(?)
+};
+
 export class RepositoryManager extends BaseManager {
 	// TODO: Add methods for repository-specific actions. E.g. creating a
 	// new repository.
+
+	async readAll(): Promise<PrismicRepository[]> {
+		const url = new URL("/repositories", APIEndpoints.PrismicUser);
+		const res = await this._fetch({ url });
+		const json = await res.json();
+
+		if (res.ok) {
+			const { value: repositories, error } = decode(
+				t.array(PrismicRepository),
+				json,
+			);
+
+			if (error) {
+				throw new Error(
+					`Failed to decode repositories: ${error.errors.join(", ")}`,
+				);
+			}
+
+			return repositories;
+		} else {
+			throw new Error(`Failed to read repositories`, { cause: json });
+		}
+	}
+
+	// Should this be in manager? It's one of the few sync method
```

Comment:
Could we add a "TODO:" at the start of the comment if we don't have a decision? It'll be helpful later when we can do a project-wide search for "TODO".

Reply:
Sure!